magma/fromverilog.py

```diff
@@ -33,7 +34,16 @@ def visit_ModuleDef(self, node):
         self.nodes.append(node)
         return node
 
-def get_type(io):
+def convert(input_type, target_type):
+    if isinstance(input_type, BitKind) and \
+       isinstance(target_type, ClockKind) and \
+       input_type.direction == target_type.direction:
```

Comment:
A more general check might be that they're both instances of `_Bitkind`? That way, any sub type of `_Bit` can be converted to another sub type of `_Bit`.

Reply:
Also, we could do similar logic for array -> bits/uint/sint (check that they're both sub types of array)

Reply:
Done and tested.